data-processing-lib/spark/src/data_processing_spark/runtime/spark/transform_orchestrator.py

```diff
@@ -0,0 +1,135 @@
+
+
+def orchestrate(
+    runtime_config: TransformRuntimeConfiguration,
+    execution_config: SparkTransformExecutionConfiguration,
+    data_access_factory: DataAccessFactoryBase,
+) -> int:
+    """
+    orchestrator for transformer execution
+    :param execution_config: orchestrator configuration
+    :param data_access_factory: data access factory
+    :param runtime_config: transformer runtime configuration
+    :return: 0 - success or 1 - failure
+    """
+    start_time = time.time()
+    start_ts = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    logger.info(f"orchestrator started at {start_ts}")
+    # create data access
+    data_access = data_access_factory.create_data_access()
+    if data_access is None:
+        logger.error("No DataAccess instance provided - exiting")
+        return 1
+    # initialize Spark
+    conf = (SparkConf().setAppName(runtime_config.get_name())
+            .set('spark.driver.host', '127.0.0.1'))
+    sc = SparkContext(conf=conf)
+    execution_params = sc.broadcast(runtime_config.get_input_params())
+    transform_class = sc.broadcast(runtime_config.get_transform_class())
+    daf = sc.broadcast(data_access_factory)
+
+    def process_partition(iterator):
+        """
+        process partitions
+        :param iterator: iterator of records
+        :return:
+        """
+        # create data access
+        d_access = daf.value.create_data_access()
+        # local statistics dictionary
+        statistics = {}
+        # create file processor
+        file_processor = SparkTransformFileProcessor(d_access=d_access, transform_params=execution_params.value,
+                                                     statistics=statistics, transform_class=transform_class.value)
+        first = True
+        for f in iterator:
+            # for every file
+            if first:
+                logger.debug(f"partition {f}")
+                # create transform with partition number
+                file_processor.create_transform(partition=int(f[1]))
+                first = False
+            # process file
+            file_processor.process_file(f_name=f[0])
+        # flush
+        file_processor.flush()
+        # return partition's statistics
+        return list(statistics.items())
+
+    try:
+        # Get files to process
+        files, profile, retries = data_access.get_files_to_process()
+        if len(files) == 0:
+            logger.error("No input files to process - exiting")
+            return 0
+        logger.info(f"Number of files is {len(files)}, source profile {profile}")
+        # process data
+        logger.debug("Begin processing files")
+        # process files split by partitions
+        logger.debug(f"parallelization {execution_config.parallelization}")
+        if execution_config.parallelization > 0:
+            stats_rdd = (sc.parallelize(files, execution_config.parallelization)
+                         .zipWithIndex().mapPartitions(process_partition))
+        else:
```

Comment:
what does it mean for paralleleization to be <= 0 ?  and do you mean > 1?

Reply:
Its explained in the documentation. if it is < 0, we use default. We really should use default everywhere. Its just a back door if you want to overwrite it